src/core/structure/sheet.ts

```diff
@@ -58,24 +58,30 @@ export default class Sheet {
 
   setCellAt(colPos: number, rowPos: number, value: string): CellInfo {
     const position = this.initializePosition(colPos, rowPos);
-    return this.setCell(position.columnKey!, position.rowKey!, value);
+    return this.setCell(position.columnKey!, position.rowKey!, value)!;
   }
 
-  setCell(colKey: ColumnKey, rowKey: RowKey, formula: string): CellInfo {
+  setCell(colKey: ColumnKey, rowKey: RowKey, formula: string): CellInfo | null {
+    if (!this.columns.has(colKey) || !this.rows.has(rowKey)) {
+      return null;
+    }
+
     let cell = this.getCell(colKey, rowKey);
-    const colIndex = this.getColumnIndex(colKey)!;
-    const rowIndex = this.getRowIndex(rowKey)!;
 
-    if (!cell) {
+    if (formula == "" && !cell) return null;
+    if (!cell && formula != "") {
       cell = this.createCell(colKey, rowKey, formula);
```

Comment:
Rewrite in simplier form (same logic)

```suggestion
    if (!cell) {
      if (formula == "") return null;
      cell = this.createCell(colKey, rowKey, formula);
    }
```

Reply:
@Fluglow any comment on this? It doesn't seem like this change went through.